### PR TITLE
Add User model, install devise gem, install rspec gem

### PR DESCRIPTION
**Pull request by SofiiaHlushchak**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -5,32 +5,26 @@ ruby '2.7.2'
...
-# Use Redis adapter to run Action Cable in production
-# gem 'redis', '~> 4.0'
-# Use Active Model has_secure_password
-# gem 'bcrypt', '~> 3.1.7'
-
-# Use Active Storage variant
-# gem 'image_processing', '~> 1.2'
+gem 'devise'
+gem 'carrierwave'
+gem 'kaminari'
```
Comment (matsevich) [inline Gemfile]: don't add gems that you will not implement in the current PR

File: config/database.yml [hunk 2]
```diff
@@ -6,10 +6,14 @@ default: &default
 development:
   <<: *default
   database: giftify_development
+  username: cassini
+  password: asd1234
```
Comment (matsevich) [inline config/database.yml]: use dotenv gem for encrypting your project credentials

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+  it 'is invalid with incorrect data' do
+    expect(user.invalid?).to be_truthy
+  end
+
+  it 'is valid with correct data' do
+    user.password = '123456'
+
+    expect(user.valid?).to be_truthy
+  end
```
Comment (manastyretskyi) [inline spec/models/user_spec.rb]: [shoulda-matchers](https://github.com/thoughtbot/shoulda-matchers)

File: spec/factories/user.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+FactoryBot.define do
+  factory :user do
+
+  end
+end
```
Comment (matsevich) [inline spec/factories/user.rb]: don't leave empty and not needed files, and if they will be needed in a separate task - then the appropriate dev will create them in his task

File: spec/models/user_spec.rb (hunk 3 above)
Comment (matsevich) [inline spec/models/user_spec.rb:7]: use shoulda-matchers for validation specs

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  describe 'validations' do
+    it { should validate_presence_of :first_name }
+    it { should validate_presence_of :last_name }
```
Comment (manastyretskyi) [inline spec/models/user_spec.rb]: краще такий синтаксис
```suggestion
    it { is_expected.to validate_presence_of :first_name }
    it { is_expected.to validate_presence_of :last_name }
```

File: db/migrate/20210521131920_create_users.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
+class CreateUsers < ActiveRecord::Migration[6.1]
+  def change
+    create_table :users do |t|
+      t.string    :first_name
+      t.string    :last_name
+      t.datetime  :birthday
```
Comment (manastyretskyi) [inline db/migrate/20210521131920_create_users.rb:6]: це напевно має бути не `datetime`, а `date`

**Pull request merged**